File: c#/Clope/MathSupport.cs
using System.Collections.Generic;

namespace Clope
{
    internal class MathSupport
    {
        public double Repulsion { get; }
        private readonly List<double> _preCalculatedWpowR;

        public double GetPreCalculatedWpowR(int w)
        {
            if (w < _preCalculatedWpowR.Count) return _preCalculatedWpowR[w];
            for (var i = _preCalculatedWpowR.Count; i <= w; i++)
                _preCalculatedWpowR.Add(System.Math.Pow(i, Repulsion));
            return _preCalculatedWpowR[w];
        }

        public MathSupport(double repulsion)
        {
            Repulsion = repulsion;
            _preCalculatedWpowR = new List<double>();
        }
    }
}

File: Typescript/src/clope/MathSupport.ts
export default class MathSupport {
    public readonly Repulsion: number;
    private PrecalculatedWPowR: number[];
    constructor(r: number) {
        this.Repulsion = r;
        this.PrecalculatedWPowR = [];
    }
    private Prepare(w: number): void {
        for (let i = this.PrecalculatedWPowR.length; i <= w; i++)
            this.PrecalculatedWPowR.push(Math.pow(i, this.Repulsion))
    }
    public GetWPowR(w: number) {
        if (w >= this.PrecalculatedWPowR.length) this.Prepare(w);
        return this.PrecalculatedWPowR[w];
    }
}